recursive_art.py

```diff
@@ -20,9 +21,69 @@ def build_random_function(min_depth, max_depth):
         (See the assignment writ-eup for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
 
+    depth = random.randint(min_depth, max_depth)
+
+    seed_function = []
+    seed_generation = 2**depth
+
+    for i in range(seed_generation):
+        xory_number = random.randint(1,2)
+
+        if xory_number == 1:
+            seed_function.append(["x"])
+
+        if xory_number == 2:
+            seed_function.append(["y"])
+
+    transient_function = seed_function
+
+    while depth > 0:
+
+        resulting_function = []
+        seed_index = 0
+
+        for i in range(2**(depth-1)):
+            function_number = random.randint(3,8)
+
+            if function_number == 3:
+                chunk = ["cos_pi"]
+                chunk.append(transient_function[seed_index])
+                seed_index += 1
+
+            elif function_number == 4:
+                chunk = ["sin_pi"]
+                chunk.append(transient_function[seed_index])
+                seed_index += 1
+
+            elif function_number == 5:
+                chunk = ["prod"]
+                chunk.append(transient_function[seed_index])
+                chunk.append(transient_function[seed_index+1])
+                seed_index += 2
+
+            elif function_number == 6:
+                chunk = ["avg"]
+                chunk.append(transient_function[seed_index])
+                chunk.append(transient_function[seed_index+1])
+                seed_index += 2
+
+            elif function_number == 7:
+                chunk = ["abs"]
+                chunk.append(transient_function[seed_index])
+                seed_index += 1
+
+            elif function_number == 8:
+                chunk = ["cubic"]
+                chunk.append(transient_function[seed_index])
+                seed_index += 1
+
+            resulting_function.append(chunk)
+
+        transient_function = resulting_function
+        depth = depth - 1
+
+    return resulting_function[0]
```

Comment:
I think you did quite a good job with not using recursion.
However, sometimes using recursion might make code more concise.
I have one suggestion for the variable names - Use 'root' for initial node for the graph, use 'temp' instead of 'transient'. The name 'seed' is usually used for fixing randomness of random functions.